File: app/src/main/java/com/kunzisoft/keepass/database/PwIconStandard.java
/*
 * Copyright 2017 Brian Pellin, Jeremy Jamet / Kunzisoft.
 *     
 * This file is part of KeePass DX.
 *
 *  KeePass DX is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  KeePass DX is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with KeePass DX.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.kunzisoft.keepass.database;

import android.os.Parcel;

public class PwIconStandard extends PwIcon {
	private final int iconId;

	public static final int UNKNOWN = -1;
	public static final int KEY = 0;
	public static final int TRASH = 43;
	public static final int FOLDER = 48;

	public PwIconStandard() {
		this.iconId = KEY;
	}

	public PwIconStandard(int iconId) {
		this.iconId = iconId;
	}

    public PwIconStandard(PwIconStandard icon) {
        this.iconId = icon.iconId;
    }

	protected PwIconStandard(Parcel in) {
		super(in);
		iconId = in.readInt();
	}

	@Override
	public boolean isUnknown() {
	    return iconId == UNKNOWN;
    }

    public int getIconId() {
        return iconId;
    }

    @Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeInt(iconId);
	}

	public static final Creator<PwIconStandard> CREATOR = new Creator<PwIconStandard>() {
		@Override
		public PwIconStandard createFromParcel(Parcel in) {
			return new PwIconStandard(in);
		}

		@Override
		public PwIconStandard[] newArray(int size) {
			return new PwIconStandard[size];
		}
	};

	@Override
	public boolean isMetaStreamIcon() {
		return iconId == 0;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + iconId;
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		PwIconStandard other = (PwIconStandard) obj;
		if (iconId != other.iconId)
			return false;
		return true;
	}
}

File: app/src/main/java/com/kunzisoft/keepass/database/element/PwIconStandard.kt
/*
 * Copyright 2019 Brian Pellin, Jeremy Jamet / Kunzisoft.
 *     
 * This file is part of KeePass DX.
 *
 *  KeePass DX is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  KeePass DX is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with KeePass DX.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.kunzisoft.keepass.database.element

import android.os.Parcel
import android.os.Parcelable

class PwIconStandard : PwIcon {

    constructor() {
        this.iconId = KEY
    }

    constructor(iconId: Int) {
        this.iconId = iconId
    }

    constructor(icon: PwIconStandard) {
        this.iconId = icon.iconId
    }

    constructor(parcel: Parcel) {
        iconId = parcel.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(iconId)
    }

    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = prime * result + iconId
        return result
    }

    override fun equals(other: Any?): Boolean {
        if (this === other)
            return true
        if (other == null)
            return false
        if (other !is PwIconStandard) {
            return false
        }
        return iconId == other.iconId
    }

    override val iconId: Int

    override val isUnknown: Boolean
        get() = iconId == UNKNOWN_ID

    override val isMetaStreamIcon: Boolean
        get() = iconId == 0

    companion object {

        const val KEY = 0
        const val TRASH = 43
        const val FOLDER = 48

        @JvmField
        val CREATOR: Parcelable.Creator<PwIconStandard> = object : Parcelable.Creator<PwIconStandard> {
            override fun createFromParcel(parcel: Parcel): PwIconStandard {
                return PwIconStandard(parcel)
            }

            override fun newArray(size: Int): Array<PwIconStandard?> {
                return arrayOfNulls(size)
            }
        }
    }
}
